### ISSUE TITLE
[BUG] incorrect lambda code file path for OpenAPI PUT/POST http methods

### ISSUE THREAD

**Issue by kenrictractable** [aws/aws-pdk#606]
### Describe the bug

I'm using OpenAPI in `TypeSafeApiProject` to generate the API, but for `POST` and `PUT` requests the generated code adds a `Operation` string to the `operationId` in the generated function in the infrastructure package, as well as generating file paths with `-operation` added for `cdk` to load.

### Expected Behavior

I don't expect `Operation` to be added in the generated code in infrastructure package, as well as `-operation` added in the file path for the lambda code.

### Current Behavior

OpenAPI definition:
```
openapi: 3.0.3
info:
  version: 1.0.0
  title: test-project
paths:
  /test:
    post:
      operationId: postTest
      x-handler:
        language: typescript
      requestBody:
        required: true
        content:
          application/json:
            schema:
              type: object
              properties:
                field1:
                  type: string
                field2:
                  type: string
      responses:
        200:
          description: OK
          content:
            'application/json':
              schema:
                $ref: '#/components/schemas/TestResponseContent'
```

Generated code under `generated/infrastructure/typescript/lib/function.js`
```
/**
 * Lambda function construct which points to the typescript implementation of PostTestOperation
 */
class PostTestOperationFunction extends aws_lambda_1.Function {
    constructor(scope, id, props) {
        super(scope, id, {
            runtime: aws_lambda_1.Runtime.NODEJS_18_X,
            handler: "index.handler",
            code: aws_lambda_1.Code.fromAsset(path.resolve(__dirname, "..", "../../../handlers/typescript/dist/lambda", "post-test-operation")),
            tracing: aws_lambda_1.Tracing.ACTIVE,
            timeout: aws_cdk_lib_1.Duration.seconds(30),
            ...props,
        });
    }
}
exports.PostTestOperationFunction = PostTestOperationFunction;
```

The generated code has `-operation` added to the file path, and this fails to build because the wrapper code that gets generated is `handlers/typescript/src/post-test.ts`, not `handlers/typescript/src/post-test-operation.ts`

### Reproduction Steps

Define a `TypeSafeApiProject` using `ModelLanguage.OPENAPI` and define a `POST` or a `PUT` HTTP method for an endpoint.

### PDK version used

0.22.17

### What languages are you seeing this issue on?

Typescript

### Environment details (OS name and version, etc.)

mac Sonoma 14.0

**Comment by cogwirrel** [comment]
Thanks for raising this!

This seems to be due to this part of OpenAPI Generator: https://github.com/OpenAPITools/openapi-generator/blob/77687a8ea59f97519731f345308d3615e2f6e500/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptFetchClientCodegen.java#L779-L781

As a workaround, the correct handler functions are generated if you use a reference, eg:

```yaml
openapi: 3.0.3
info:
  version: 1.0.0
  title: Test
paths:
  /hello:
    post:
      operationId: postTest
      x-handler:
        language: typescript
      requestBody:
        required: true
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/PostTestRequestContent'
      responses:
        200:
          description: Successful response
          content:
            'application/json':
              schema:
                $ref: '#/components/schemas/PostTestResponseContent'
        500:
          description: An internal failure at the fault of the server
          content:
            'application/json':
              schema:
                $ref: '#/components/schemas/InternalFailureErrorResponseContent'
        400:
          description: An error at the fault of the client sending invalid input
          content:
            'application/json':
              schema:
                $ref: '#/components/schemas/BadRequestErrorResponseContent'
        403:
          description: An error due to the client not being authorized to access the resource
          content:
            'application/json':
              schema:
                $ref: '#/components/schemas/NotAuthorizedErrorResponseContent'
components:
  schemas:
    PostTestRequestContent:
      type: object
      properties:
        field1:
          type: string
        field2:
          type: string
    PostTestResponseContent:
      type: object
      properties:
        message:
          type: string
      required:
        - message
    InternalFailureErrorResponseContent:
      type: object
      properties:
        message:
          type: string
      required:
        - message
    BadRequestErrorResponseContent:
      type: object
      properties:
        message:
          type: string
      required:
        - message
    NotAuthorizedErrorResponseContent:
      type: object
      properties:
        message:
          type: string
      required:
        - message
```

I'll have a look at whether we can support the inline case! :)

**Comment by cogwirrel** [comment]
From a bit more digging, unfortunately the operation ID will always get `Operation` added as a suffix in generated typescript code for inline request bodies as this is the TypeScript OpenAPI generator's behaviour and would be tricky to avoid. So the best thing to do to avoid the `Operation` suffix appearing in the generated code altogether is to use `$ref` as mentioned above.

However, there is a bug in Type Safe API too where we point to the file that doesn't exist, where we add `-operation` to the handler file name which is incorrect. This is due to the handler filename using `{{nickname}}` but the reference to it using `{{operationIdCamelCase}}` - the former doesn't have the suffix but the latter does!

I'll send a PR to fix this soon.

**Comment by cogwirrel** [comment]
Fixed in [0.22.18](https://github.com/aws/aws-pdk/releases/tag/v0.22.18) :)
